bin/github-review.sh

```diff
@@ -45,12 +45,40 @@ if [[ -z $chainid ]]; then
     echo "ERROR: Chain ID not specified as per the PR Template" 1>&2
     exit 1
 fi
-chainInfo="https://raw.githubusercontent.com/ethereum-lists/chains/refs/heads/master/_data/chains/eip155-$chainid.json"
-rpc="$(curl -sfL "$chainInfo" | jq -r '.rpc[0]')"
-if [[ -z $rpc ]]; then
-    echo "ERROR: RPC not fetched correctly from the ethereum-lists" 1>&2
+
+# Fetch RPC from DefiLlama's chainlist
+chainlist_response=$(curl -sfL 'https://chainlist.org/rpcs.json' || echo "CURL_ERROR:$?")
+if [[ $chainlist_response == CURL_ERROR:* ]]; then
+    error_code=${chainlist_response#CURL_ERROR:}
+    echo "ERROR: Failed to fetch chainlist from DefiLlama (curl error $error_code)" 1>&2
+    exit 1
+fi
+
+# Parse the chainlist response
+if ! chainlist=$(echo "$chainlist_response" | jq -e '.'); then
+    echo "ERROR: RPC not found for chain ID $chainid in DefiLlama's ChainList" 1>&2
+    exit 1
+fi
+
+# Extract RPC for the specified chain ID using jq
+# First check if the chain exists and has an RPC
+chain_exists=$(echo "$chainlist" | jq -r "map(select(.chainId == $chainid and (.rpc | length > 0))) | length > 0")
+if [[ $chain_exists != "true" ]]; then
+    echo "ERROR: RPC not found for chain ID $chainid in DefiLlama's chainlist" 1>&2
     exit 1
 fi
+
+# Then get the RPC
+if ! rpc=$(echo "$chainlist" | jq -r ".[] | select(.chainId == $chainid) | .rpc[0].url"); then
```

Comment:
I don't know how much you can do with `jq` script injection. But to be safe, I would suggest:


```suggestion
if ! rpc=$(echo "$chainlist" | jq --arg C "$chainid" -r '.[] | select((.chainId | tostring) == $C) | .rpc[0].url'); then
```